tests/e2e/script_test.go

```diff
@@ -444,8 +456,16 @@ func createKubeconfigFromRestConfig(restConfig *rest.Config) ([]byte, error) {
 	return clientcmd.Write(clientConfig)
 }
 
+type cliMockOptions struct {
+	changeStateIntoSpecOptions
+}
+
+type changeStateIntoSpecOptions struct {
+	getObjectResult []byte
+}
+
 // runCLI runs the config-connector CLI tool with the specified arguments
-func runCLI(h *create.Harness, args []string, uniqueID string, baseOutputPath string) {
+func runCLI(h *create.Harness, args []string, uniqueID string, baseOutputPath string, mockOptions *cliMockOptions) {
```

Comment:
Can we wrap the arguments in a struct for args?

Reply:
For sure we can, though I picked using an extra struct because I lean towards having the main powertool logic and testing logic separated.

Here is my _biased_ take about the options we have:
1. Option 1: Use an additional struct to capture the mocking option
    *  Pro:
        1. It keeps the testing logic and the main powertool logic separated.
        2. The args in the test configuration matches the real case usage of the powertool.
    * Con:
       1. The testing logic is complicated and requires additional learning.
1. Option 2: Use args slice to capture the mocking option:
    1. Option 2.1: Pass all the arguments from args to cmd.TestInvocationOptions.
        *  Pro:
             1. It's an easier interface to construct the test case.
        * Con:
             1. The powertool itself (ExecuteFromTest()) needs to be able to handle the mocking options. This mixes the testing logic into the powertool and increases the complexity of the tool.
             4. It may be confusing to the developers because the test case and the real usage of the powertool take different arguments.
    1. Option 2.2: Pass all the arguments except the ones for the mocking options to cmd.TestInvocationOptions.
        *  Pro:
             1. It's an easier interface to construct the test case.
        * Con:
             1. Special keywords need to be reserved for testing purpose and additional logic needs to be done in runCLI() to handle the mocking option related args.

Again, this is probably quite hacky and I'm open to any improvement ideas!